lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1957,6 +1957,108 @@ def plot(self, xs, ys, *args, zdir='z', **kwargs):
+            the *True* position remain unfilled due to the adjacent *False*
+            values.
+
+        mode : {'auto', 'quad', 'polygon'}, default: 'auto'
+            The fill mode.
+            If 'auto' and all the coordinates for the x, y, or z input pairs
+            are the same, then the points lie on a plane and 'polygon' is used.
+            Otherwise, 'auto' defaults to 'quad'.
+            If 'quad', then a separate quadrilateral polygon is created for
+            each pair of subsequent points in the 1st and 2nd lines. This is
+            more flexible, but may be slower and result in rendering artifacts.
+            If 'polygon', then the 1st and 2nd lines are connected to form a
+            single polygon, which may be faster and render more cleanly for
+            simple shapes (eg, for filling between two lines that lie on a
+            single plane).
+
+        **kwargs
+            All other keyword arguments are passed on to `.Poly3DCollection`.
+
+        Returns
+        -------
+        `.Poly3DCollection`
+            A `.Poly3DCollection` containing the plotted polygons.
+
+        """
+        _api.check_in_list(['auto', 'quad', 'polygon'], mode=mode)
+
+        had_data = self.has_data()
+        x1, y1, z1, x2, y2, z2 = cbook._broadcast_with_masks(x1, y1, z1, x2, y2, z2)
+        if mode == 'auto':
+            if np.all(x1 == x2) or np.all(y1 == y2) or np.all(z1 == z2):
+                mode = 'polygon'
+            else:
+                mode = 'quad'
+
+        if where is None:
+            where = True
+        else:
+            where = np.asarray(where, dtype=bool)
+            if where.size != x1.size:
+                raise ValueError(f"where size ({where.size}) does not match "
+                                 f"size ({x1.size})")
+        where = where & ~np.isnan(x1)  # NaNs were broadcast in _broadcast_with_masks
+
+        polys = []
+        for idx0, idx1 in cbook.contiguous_regions(where):
+            x1slice = x1[idx0:idx1]
+            y1slice = y1[idx0:idx1]
+            z1slice = z1[idx0:idx1]
+            x2slice = x2[idx0:idx1]
+            y2slice = y2[idx0:idx1]
+            z2slice = z2[idx0:idx1]
+
+            if not len(x1slice):
+                continue
+
+            if mode == 'quad':
+                for i in range(len(x1slice) - 1):
+                    poly = [(x1slice[i], y1slice[i], z1slice[i]),
+                            (x1slice[i+1], y1slice[i+1], z1slice[i+1]),
+                            (x2slice[i+1], y2slice[i+1], z2slice[i+1]),
+                            (x2slice[i], y2slice[i], z2slice[i])]
+                    polys.append(poly)
+            elif mode == 'polygon':
+                poly = []
+                for i in range(len(x1slice)):
+                    poly.append((x1slice[i], y1slice[i], z1slice[i]))
+                for i in range(len(x2slice) - 1, -1, -1):
+                    poly.append((x2slice[i], y2slice[i], z2slice[i]))
```

Comment:
We should try to vectorize this if easy enough to do. I think the following will work.
```suggestion
                # Stack the top points in a line
                top = np.column_stack((x1slice, y1slice, z1slice))
                # Stack the second slice in reverse order
                bottom = np.column_stack((x2slice[::-1], y2slice[::-1], z2slice[::-1]))
                # Concatenate the lines into a single poly
                poly = np.concatenate((top, bottom), axis=0)
```

Reply:
I like these vectorizations, done